curated_transformers/generation/auto_generator.py

```diff
@@ -0,0 +1,70 @@
+from typing import Dict, Optional, Type, cast
+
+import torch
+
+from curated_transformers.generation.default_generator import DefaultGenerator
+from curated_transformers.generation.hf_hub import FromHFHub
+
+from ..quantization import BitsAndBytesConfig
+from ..util.auto_model import AutoModel
+from .dolly_v2 import DollyV2Generator
+from .falcon import FalconGenerator
+from .generator_wrapper import GeneratorWrapper
+
+# For the time being, we enable support for a generator on a case-by-case basis.
+# In the future we might defer all unknown generators to DefaultGenerator.
+GENERATOR_MAP: Dict[str, Type[FromHFHub]] = {
+    "dolly-v2": DollyV2Generator,
+    "falcon": FalconGenerator,
+    "llama": DefaultGenerator,
+}
+
+
+class AutoGenerator(AutoModel[GeneratorWrapper]):
+    """Causal LM generator loaded from the Hugging Face Model Hub.
+
+    **NOTE** - This class can currently only be used with the following models:
+        - Models based on Dolly v2 (contain ``dolly-v2``` in the name).
+        - Models based on Falcon (contain ``falcon`` in the name).
+        - Models based on LLaMA (contain ``llama`` in the name).
+    """
```

Comment:
```suggestion
    """
     .. highlights::
    	Causal LM generator loaded from the Hugging Face Model Hub.

        .. attention::
          	This class can currently only be used with the following models:
          	 - Models based on Dolly v2 (contain ``dolly-v2`` in the name).
          	 - Models based on Falcon (contain ``falcon`` in the name).
          	 - Models based on LLaMA (contain ``llama`` in the name).
    """
```
To keep in line with the upcoming docs PR.

Reply:
I am not sure I understand these? `highlights` is for block quotes?

https://docutils.sourceforge.io/docs/ref/rst/directives.html#highlights

Reply:
As discussed elsewhere, we can skip the `.. highlights::` annotation in favour of a global Sphix config setting.

Reply:
Added the attention block.